dashboard/app/models/concept.rb

```diff
@@ -56,11 +53,16 @@ def self.setup
   def self.setup_with_concepts(concepts_by_index)
     videos_by_concept = Video.where(key: concepts_by_index).index_by(&:key)
     concepts = concepts_by_index.map.with_index(1) do |concept, id|
-      {id: id, name: concept, video_id: videos_by_concept[concept]&.id}
+      {id: id, name: concept, video_key: videos_by_concept[concept]&.key}
     end
     transaction do
       reset_db
       Concept.import! concepts
     end
   end
+
+  def related_video
+    @@related_video ||= {}
+    @@related_video[video_key + ":" + I18n.locale.to_s] ||= Video.current_locale.find_by_key(video_key) unless video_key.nil?
```

Comment:
I'd recommend using an actual cache here if this value is important to cache; class variables are not thread-safe, and tend to cause problems in server environments

Reply:
Hmm, TIL. Should we track updating all the other places we use class vars to cache lookups?

https://github.com/code-dot-org/code-dot-org/blob/e13cb375c14f0a4ef7f89bdff4fd89e5618f080b/dashboard/app/models/game.rb#L31

Reply:
This model already uses class variables like this, as well as other model such as Level. If we aren't comfortable with this usage of class variables, I'd rather change at least this whole model in a different PR.

Reply:
I've got #27961 out to fix the other instances of this in this file and updated this PR to fix this new function.